pyomo/solvers/tests/checks/test_CPLEXDirect.py

```diff
@@ -255,13 +255,13 @@ def test_soln_limit_mip(self):
 
             opt.options['mip_limits_solutions'] = 1
             results = opt.solve(model)
-
             self.assertEqual(results.solver.status,
                              SolverStatus.aborted)
             self.assertEqual(results.solver.termination_condition,
                              TerminationCondition.unknown)
             self.assertEqual(model.solutions[0].status,
                              SolutionStatus.stoppedByLimit)
+            self.assertGreater(results.solver.deterministic_time, 0.0)
```

Comment:
Perhaps a slightly stronger test would be to assert on the value itself?

I think it would be enough to just duplicate this test and pass in `opt.options['dettimelimit'] = 10` instead of `mip_limits_solutions`. Depending on how CPLEX behaves, we may need to assert that `results.solver.deterministic_time` is within a few ticks of 10; I'm not entirely sure whether it does stop exactly at the limit or whether it will perform some small cleanup as well.

Reply:
As you said, `results.solver.deterministic_time` returns `0.0009851455688476562` when I set `opt.options['dettimelimit'] = 0.0001`

Reply:
Hmmm that's interesting... Maybe it would help to increase this to 10? 1 tick is actually a very short amount of time because I can see from the logs that CPLEX typically runs at several 100's of ticks _per second_.

Reply:
Also this test is failing for a couple of the pipelines so that proves that the `deterministic_time` is non-deterministic _across_ platforms. As described above, I think using a much larger magnitude of 10 and asserting within a tolerance of 9-11 ticks should alleviate this.

Reply:
Optimal solution can be found in 0.0015420913696289062 ticks for this super tiny problem, that's why I set 0.0001 (hoping to hit the time limit and get `results.solver.deterministic_time=0.0001`. The best solution I can think of is to assert within a tolerance of 0-0.0001 ticks under the setting of `opt.options['dettimelimit'] = 0.0001`

Reply:
Also, the test failed in the pipelines with `0.0029306411743164062 != 0.0009851455688476562`. (`0.0009851455688476562` was the result I got locally.) I think deterministic time might also be CPLEX version dependent which is something we cannot control. I prefer not to assert any UB.